plugins/derive/g_derive.mlg

```diff
@@ -20,10 +20,30 @@ DECLARE PLUGIN "coq-core.plugins.derive"
 
 let classify_derive_command _ = Vernacextend.(VtStartProof (Doesn'tGuaranteeOpacity,[]))
 
+(* BEGIN HACK TO DECLARE SuchThat AS KEYWORD *)
+let wit_suchthat =
+ (Genarg.create_arg "suchthat" : string Genarg.uniform_genarg_type)
+
+let suchthat = Pcoq.create_generic_entry2 "binders" (Genarg.rawwit wit_suchthat)
```

Comment:
Why do we need a genarg? Can't we just
```suggestion
let suchthat : unit Pcoq.Entry.t = Pcoq.Entry.make "suchthat"
```
? and use `"SuchThat"` directly in the VERNAC COMMAND EXTEND

Reply:
It seems that `VERNAC COMMAND EXTEND` still requires a `wit`. Without it, this gives `Unbound value wit_suchthat`.

Reply:
> use "SuchThat" directly in the VERNAC COMMAND EXTEND

In theory yes but this would require supporting keyword in `VERNAC COMMAND EXTEND`, which is not the case as far as I could see. And we need "SuchThat" to be a keyword since otherwise it would be parsed as one of the names defined by `Derive`.

Reply:
How does this use of `"for"` work then? https://github.com/coq/coq/blob/3a19736f7377d5bd755662800954913301858bbb/plugins/funind/g_indfun.mlg#L292
(`for` is a keyword (used in `fix ... for x`))

Reply:
This [line](https://github.com/coq/coq/blob/3a19736f7377d5bd755662800954913301858bbb/parsing/g_constr.mlg#L333), without `IDENT`, is e.g. one which forces `"foo"` to be a keyword.

Reply:
But in `VERNAC COMMAND EXTEND`, a `"foo"` is by default interpreted as `IDENT "foo"`.

Reply:
my suggestion is that we have
~~~
{
(* dummy entry to make "SuchThat" a keyword *)
let suchthat : unit Pcoq.Entry.t = Pcoq.Entry.make "suchthat"
}

(* dummy grammar to make "SuchThat" a keyword *)
GRAMMAR EXTEND Gram
  GLOBAL: suchthat;
    suchthat:
    [ [ x = "SuchThat" -> { x } ] ];
END


VERNAC COMMAND EXTEND ...
... "Derive" open_binders(bl) "SuchThat" ...
~~~
does it not work?

Reply:
Yes, that probably works, but then, rather than doing a dummy grammar, it would probably be clearer to (re)add a function `declare_token` and to explicitly declare `SuchThat` as a token.

Reply:
I simplified by registering explicitly `"SuchThat"` as keyword.